### ISSUE TITLE
failing in umbrella project on elixir 1.15 but not on elixir 1.14

### ISSUE THREAD

**Issue by moperacz** [elixir-lang/ex_doc#1769]
I've created clean project using `mix phx.new testing_ex_doc --umbrella` added:

```elixir
name: "MyApp",
      source_url: "https://github.com/USER/PROJECT",
      homepage_url: "http://YOUR_PROJECT_HOMEPAGE",
      docs: [
        # The main page in the docs
        main: "MyApp",
        # logo: "path/to/logo.png",
        extras: ["README.md"]
      ]
```
to project's root `mix.exs` also added `{:ex_doc, "~> 0.27", only: :dev, runtime: false}` to deps.

the result in elixir 1.15 is:

```text
mix docs
==> earmark_parser
Compiling 1 file (.yrl)
Compiling 2 files (.xrl)
Compiling 3 files (.erl)
Compiling 46 files (.ex)
Generated earmark_parser app
==> nimble_parsec
Compiling 4 files (.ex)
Generated nimble_parsec app
==> makeup
Compiling 44 files (.ex)
Generated makeup app
==> makeup_elixir
Compiling 6 files (.ex)
Generated makeup_elixir app
==> makeup_erlang
Compiling 3 files (.ex)
Generated makeup_erlang app
==> ex_doc
Compiling 25 files (.ex)
Generated ex_doc app
==> testing_ex_doc
Generated testing_ex_doc app
==> testing_ex_doc_web
Generated testing_ex_doc_web app
Generating docs...
** (UndefinedFunctionError) function :string_lexer.string/1 is undefined (module :string_lexer is not available)
    (earmark_parser 1.4.34) :string_lexer.string(~c"A convenience function for SQL-based repositories that forces all connections in the")
    (earmark_parser 1.4.34) lib/earmark_parser/helpers/leex_helpers.ex:20: EarmarkParser.Helpers.LeexHelpers.tokenize/2
    (earmark_parser 1.4.34) lib/earmark_parser/helpers/lookahead_helpers.ex:17: EarmarkParser.Helpers.LookaheadHelpers.opens_inline_code/1
    (earmark_parser 1.4.34) lib/earmark_parser/parser.ex:654: EarmarkParser.Parser._inline_or_text?/2
    (earmark_parser 1.4.34) lib/earmark_parser/parser.ex:482: EarmarkParser.Parser._consolidate_para/4
    (earmark_parser 1.4.34) lib/earmark_parser/parser.ex:211: EarmarkParser.Parser._parse/4
    (earmark_parser 1.4.34) lib/earmark_parser/parser.ex:64: EarmarkParser.Parser.lines_to_blocks/3
    (earmark_parser 1.4.34) lib/earmark_parser/parser.ex:56: EarmarkParser.Parser.parse_lines/3
```

but when i switch to elixir 1.14 i've got:
```text
 mix docs
==> earmark_parser
Compiling 3 files (.erl)
Compiling 46 files (.ex)
Generated earmark_parser app
==> file_system
Compiling 7 files (.ex)
Generated file_system app
==> decimal
Compiling 4 files (.ex)
Generated decimal app
==> mime
Compiling 1 file (.ex)
Generated mime app
==> nimble_options
Compiling 3 files (.ex)
Generated nimble_options app
==> nimble_parsec
Compiling 4 files (.ex)
Generated nimble_parsec app
===> Analyzing applications...
===> Compiling telemetry
==> telemetry_metrics
Compiling 7 files (.ex)
Generated telemetry_metrics app
===> Analyzing applications...
===> Compiling telemetry_poller
==> jason
Compiling 10 files (.ex)
Generated jason app
==> db_connection
Compiling 15 files (.ex)
Generated db_connection app
==> expo
Compiling 2 files (.erl)
Compiling 21 files (.ex)
Generated expo app
==> phoenix_pubsub
Compiling 11 files (.ex)
Generated phoenix_pubsub app
==> plug_crypto
Compiling 5 files (.ex)
Generated plug_crypto app
==> hpax
Compiling 4 files (.ex)
Generated hpax app
==> gettext
Compiling 17 files (.ex)
Generated gettext app
===> Analyzing applications...
===> Compiling ranch
==> ecto
Compiling 56 files (.ex)
Generated ecto app
==> plug
Compiling 1 file (.erl)
Compiling 40 files (.ex)
Generated plug app
==> phoenix_html
Compiling 9 files (.ex)
Generated phoenix_html app
==> phoenix_template
Compiling 4 files (.ex)
Generated phoenix_template app
==> postgrex
Compiling 68 files (.ex)
Generated postgrex app
==> makeup
Compiling 44 files (.ex)
Generated makeup app
==> makeup_elixir
Compiling 6 files (.ex)
Generated makeup_elixir app
==> makeup_erlang
Compiling 3 files (.ex)
Generated makeup_erlang app
==> ex_doc
Compiling 25 files (.ex)
Generated ex_doc app
==> ecto_sql
Compiling 25 files (.ex)
Generated ecto_sql app
==> nimble_pool
Compiling 2 files (.ex)
Generated nimble_pool app
==> castore
Compiling 1 file (.ex)
Generated castore app
==> esbuild
Compiling 3 files (.ex)
Generated esbuild app
==> tailwind
Compiling 3 files (.ex)
Generated tailwind app
==> mint
Compiling 1 file (.erl)
Compiling 19 files (.ex)
Generated mint app
==> finch
Compiling 13 files (.ex)
Generated finch app
==> websock
Compiling 1 file (.ex)
Generated websock app
===> Analyzing applications...
===> Compiling cowlib
===> Analyzing applications...
===> Compiling cowboy
===> Analyzing applications...
===> Compiling cowboy_telemetry
==> plug_cowboy
Compiling 5 files (.ex)
Generated plug_cowboy app
==> websock_adapter
Compiling 2 files (.ex)
Generated websock_adapter app
==> phoenix
Compiling 71 files (.ex)
Generated phoenix app
==> phoenix_live_view
Compiling 37 files (.ex)
Generated phoenix_live_view app
==> phoenix_live_dashboard
Compiling 36 files (.ex)
Generated phoenix_live_dashboard app
==> phoenix_live_reload
Compiling 4 files (.ex)
Generated phoenix_live_reload app
==> swoosh
Compiling 44 files (.ex)
Generated swoosh app
==> phoenix_ecto
Compiling 7 files (.ex)
Generated phoenix_ecto app
==> testing_ex_doc
Compiling 4 files (.ex)
Generated testing_ex_doc app
==> testing_ex_doc_web
Compiling 12 files (.ex)
Generated testing_ex_doc_web app
Generating docs...
warning: index.html redirects to MyApp.html, which does not exist
View "html" docs at "doc/index.html"
View "epub" docs at "doc/MyApp.epub"
```

the `yrl` and `xrl` files from `earmark_parser` were compiled but i suspect they won't load in 1.15 any idea? or maybe i should create this issue somewhere else (`earmark_parser`/`elixir-lang`)?

**Comment by moperacz** [comment]
versions i've tested:

working:
```
elixir --version
Erlang/OTP 26 [erts-14.0.2] [source] [64-bit] [smp:8:8] [ds:8:8:10] [async-threads:1] [jit]

Elixir 1.14.5 (compiled with Erlang/OTP 26)
```

failing:
```
elixir --version
Erlang/OTP 26 [erts-14.0.2] [source] [64-bit] [smp:8:8] [ds:8:8:10] [async-threads:1] [jit]

Elixir 1.15.5 (compiled with Erlang/OTP 26)
```

**Comment by josevalim** [comment]
Thank you for the report. It is an Elixir bug and I am working on a fix. :)

**Comment by josevalim** [comment]
Fixed in Elixir main and v1.15 branches.
